app/api/routers/breeze_buddy/user_accounts/handlers.py

```
@@ -0,0 +1,404 @@
+    # Cannot delete self
+    if user.id == current_user.id:
+        raise HTTPException(status_code=400, detail="Cannot delete your own account")
+
+    # Check delete permissions
+    _check_update_delete_access(current_user, user)
+
+    try:
+        success = await user_queries.delete_user(user_id, current_user.role)
+
+        if not success:
+            raise HTTPException(status_code=404, detail="User not found")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)